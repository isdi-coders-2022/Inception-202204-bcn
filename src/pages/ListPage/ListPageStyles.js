import styled from "styled-components";

const ListPageStyles = styled.div`
  display: flex;
  justify-content: center;

  padding-top: 40px;
  color: white;

  ol {
    list-style: none;
    padding: 0;
    padding-bottom: 60px;
  }
`;

export default ListPageStyles;
