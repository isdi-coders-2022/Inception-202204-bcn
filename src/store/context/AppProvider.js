import { useEffect, useReducer } from "react";
import { loadAlcoholicDrinksActionCreation } from "../actions/creador";
import appReducer from "../appReducer/appReducer";
import AppContext from "./AppContext";

const AppProvider = ({ children }) => {
  const dataBase = {
    alcoholicDrinks: [],
    nonAlcoholicDrinks: [],
    mostPopularDrinks: [],
    latestDrinks: [],
  };

  const [drinksData, dispatch] = useReducer(appReducer, dataBase);

  useEffect(() => {
    (async () => {
      const response = await fetch(
        `https://www.thecocktaildb.com/api/json/v1/1/filter.php?a=Alcoholic`
      );
      let drinksData = await response.json();

      dispatch(loadAlcoholicDrinksActionCreation(drinksData.drinks));
    })();
  }, []);

  return (
    <AppContext.Provider value={{ drinksData, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

export default AppProvider;
