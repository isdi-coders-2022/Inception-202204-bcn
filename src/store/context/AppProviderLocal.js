import { useEffect, useReducer } from "react";
import { loadUserDrinksActionCreation } from "../actions/creador";
import appReducer from "../appReducer/appReducer";
import AppContextLocal from "./AppContextLocal";

const AppProviderLocal = ({ children }) => {
  const dataBase = {
    drinks: [],
  };

  let [drinksData, dispatch] = useReducer(appReducer, dataBase);

  useEffect(() => {
    (async () => {
      const response = await fetch(`https://drinks-api.onrender.com/drinks`);
      let drinksData = await response.json();

      dispatch(loadUserDrinksActionCreation(drinksData));
    })();
  }, []);

  return (
    <AppContextLocal.Provider value={{ drinksData, dispatch }}>
      {children}
    </AppContextLocal.Provider>
  );
};

export default AppProviderLocal;
