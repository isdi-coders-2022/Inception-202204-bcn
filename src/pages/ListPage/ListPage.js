import { useContext, useEffect } from "react";
import List from "../../components/List/List";
import useApi from "../../hooks/useApi";
import AppContext from "../../store/context/AppContext";
import ListPageStyles from "./ListPageStyles";

const ListPage = () => {
  const { loadPageList } = useApi();
  const { drinksData } = useContext(AppContext);

  useEffect(() => {
    loadPageList();
  }, [loadPageList]);

  const button = "/images/favourites_icon.png";

  return (
    <ListPageStyles>
      <List
        buttonAction={"add"}
        buttonSource={button}
        drinks={drinksData.alcoholicDrinks}
      />
      {/*       <ol>
        {drinksData.alcoholicDrinks.map((drink) => {
          return (
            <SingleDrink
              item={drink}
              key={drink.idDrink}
              buttonSource={button}
              buttonAction={"add"}
            />
          );
        })}
      </ol> */}
    </ListPageStyles>
  );
};

export default ListPage;
