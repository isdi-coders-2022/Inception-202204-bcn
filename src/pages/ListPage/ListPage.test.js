import React from "react";
import ListPage from "./ListPage";
import TestRenderer from "react-test-renderer";
import "@testing-library/jest-dom";
import AppProvider from "../../store/context/AppProvider";
import AppProviderLocal from "../../store/context/AppProviderLocal";

describe("Given ListPage component", () => {
  describe("When rendered", () => {
    test("Then it should always match this snapshot", () => {
      const formularyPage = TestRenderer.create(
        <AppProvider>
          <AppProviderLocal>
            <ListPage />
          </AppProviderLocal>
        </AppProvider>
      );
      expect(formularyPage).toMatchSnapshot();
    });
  });
});
