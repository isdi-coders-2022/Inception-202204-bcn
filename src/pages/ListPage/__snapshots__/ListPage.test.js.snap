// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given ListPage component When rendered Then it should always match this snapshot 1`] = `
<div
  className="sc-dkzDqf dcbHtU"
>
  <ol />
</div>
`;
